OCLB.user.js

```diff
@@ -12,651 +12,758 @@
+                            success: function () {
+                                var resultJSON = JSON.parse(this);
+                                if (!this || this.includes('fail')) {
+                                    callback(0, 'unknown', TITLES.unknown.err_dev_id);
+                                    return;
+                                }
+
+                                if (!resultJSON.canGiveLlama) {
+                                    callback(devIDs[devName], 'already');
+                                } else if (resultJSON.canGiveLlama) {
+                                    callback(devIDs[devName], 'give');
+                                } else {
+                                    callback(devIDs[devName], 'unknown', TITLES.unknown.err_server_response);
+                                }
+                            },
+                            error: function () {
+                                callback(0, 'unknown', TITLES.unknown.err_network);
+                            },
+                        });
+                    } else {
+                        // Handle the case where CSRF token is still not found
+                        console.error("CSRF token not found.");
+                        setButtonsState(devName, 'token_miss', "Token not found! Refresh and retry..");
+                    }
+                })
+                .catch(error => {
+                    // Handle any errors that occurred during the process
+                    console.error("Error:", error);
+                });
         };
-        if (data.successText.includes('Success!')) {
-          callback('success', true);
-        } else if (data.errorText.includes('You cannot give any more llama badges to') || data.errorText.includes('Cannot give badge to this user')) {
-          callback('already', true);
-        } else if (data.errorText.includes('Badges have been given too quickly, and have tripped a spam filter')) {
-          callback('spam');
-        } else {
-          callback('error');
-          if (oclbFrame) {
-            if (data.errorText.includes('Please enter a password')) {
-              data.errorText = 'One Click Llama Button will start working after you give a Llama manually (the normal way) with the "Remember my password" option checked.';
+        var llamaButtonsToUpdate = {};
+        var askServerForStatus = function (llamaButton, devName) {
+            if ({}.hasOwnProperty.call(llamaButtonsToUpdate, devName)) {
+                llamaButtonsToUpdate[devName].push(llamaButton);
+            } else {
+                llamaButtonsToUpdate[devName] = [llamaButton];
+                getGiveMenu(devName, function (devID, className, title) {
+                    saveLastState(devName, className, title);
+                    if (devID) devIDs[devName] = devID;
+                    // if (className === 'already') storage('set', loggedInDev + '|' + devName, 0);
```

Comment:
was this commented accidentally?

Reply:
Hello, it was not accidentally. The reason it was commented because as the endpoint was mismatching, all of the llama badges were showing as unknown and it is storing in localStorage as 0. That's why it was not working fine with previous users. So, I have removed this line to not to record anything and work perfectly.